### ISSUE TITLE
longRunningRecognizeAsync in asyncRecognizeGcs doesn't work with long audio files. 

### ISSUE THREAD

**Issue by b-loved-dreamer** [googleapis/java-speech#358]
While troubleshooting this [issue](https://github.com/GoogleCloudPlatform/php-docs-samples/issues/1094) I discovered that the ```longRunningRecognizeAsync ```in this [sample](https://github.com/googleapis/java-speech/blob/master/samples/snippets/src/main/java/com/example/speech/Recognize.java) presents a similar behavior as the original issue I was looking into. 

To repro:

Use the ```asyncRecognizeGcs ``` from the [samples](https://github.com/googleapis/java-speech/blob/master/samples/snippets/src/main/java/com/example/speech/Recognize.java) and feed it a long gcs audio file (at least 30 minutes long) and run it.

The process will run for a while to render no results.
Will see a "Task was cancelled." message.

**Comment by b-loved-dreamer** [comment]
For Java LROs, you get back a Future (really an ApiFuture) which is doing polling on the operation status in the background. The "Task is cancelled" actually means that the configured timeout for polling has been reached, but is not reflective of the actual operation on the backend, which may still be running.

Found a work-around for this issue:

```
public static void asyncRecognizeGcs(String gcsUri) throws Exception {
  // Instantiates a client with GOOGLE_APPLICATION_CREDENTIALS


//Create a stubSettingsBuilder that will allow to update the polling timeout.
   SpeechStubSettings.Builder stubSettingsBuilder = SpeechStubSettings.newBuilder();

   stubSettingsBuilder
            .longRunningRecognizeOperationSettings()
            .setPollingAlgorithm(
                OperationTimedPollAlgorithm.create(
                    RetrySettings.newBuilder()
                        .setInitialRetryDelay(Duration.ofMillis(1000L))
                        .setRetryDelayMultiplier(1.5)
                        .setMaxRetryDelay(Duration.ofMillis(10000L))
                        .setInitialRpcTimeout(Duration.ZERO)
                        .setRpcTimeoutMultiplier(2.0)
                        .setMaxRpcTimeout(Duration.ZERO)
                        .setTotalTimeout(Duration.ofMillis(1000000L))/update this as needed
                        .build()));//
   
  //Construct the speechSettings
  SpeechSettings speechSettings = SpeechSettings.create(stubSettingsBuilder.build());

  try (SpeechClient speech = SpeechClient.create(speechSettings)) {//Construct the SpeechClient with the speechSettings

    // Configure remote file request for FLAC
    RecognitionConfig config =
        RecognitionConfig.newBuilder()
            .setEncoding(AudioEncoding.FLAC)
            .setLanguageCode("en-US")
            .setSampleRateHertz(16000)
            .build();
    RecognitionAudio audio = RecognitionAudio.newBuilder().setUri(gcsUri).build();

    // Use non-blocking call for getting file transcription
    OperationFuture<LongRunningRecognizeResponse, LongRunningRecognizeMetadata> response =
        speech.longRunningRecognizeAsync(config, audio);
    while (!response.isDone()) {
      System.out.println("Waiting for response...");
      Thread.sleep(10000);
    }

    List<SpeechRecognitionResult> results = response.get().getResultsList();

    for (SpeechRecognitionResult result : results) {
      // There can be several alternative transcripts for a given chunk of speech. Just use the
      // first (most likely) one here.
      SpeechRecognitionAlternative alternative = result.getAlternativesList().get(0);
      System.out.printf("Transcription: %s\n", alternative.getTranscript());
    }
  }
}
```

TLDR: In the future the defaults will basically be 24 hours on the polling for operation completion.
